park-dashboard/src/Components/StampingCenters/PassportCenters.jsx

```diff
@@ -1,20 +1,43 @@
 import React from "react";
 import { StampingList } from "./StampingList";
+import "./stamping.css";
+import stampImage from "./stampLogo.png";
 
 const PassportCenters = (centers) => {
   const centersData = StampingList(centers);
   const parkCode = centers.parkCode;
-  const center = centersData.find((c) => c.parks[0].parkCode === parkCode);
+  let stampingCenters = [];
+
+  for (let i = 0; i < centersData.length; i++) {
+    const center = centersData[i];
+    const parks = center.parks;
+
+    for (let j = 0; j < parks.length; j++) {
+      const park = parks[j];
+
+      if (park.parkCode === parkCode) {
+        stampingCenters.push(center);
+        break;
+      }
+    }
+  }
+
   return (
-    <div className="box">
-      <h2>Passport Stamping</h2>
-      {center ? (
+    <div className="passport-locations">
+      <div className="stamping-image">
+      <img className="stamp" src={stampImage} alt="Stamping-Logo" />
+      </div>
+      <div className="location-list">
+      {stampingCenters.length > 0 ? (
         <ul>
-          <li>{center.label}</li>
+          {stampingCenters.map((center) => (
+            <div className="center-name"><li key={center.label}>{center.label}</li></div>
```

Comment:
Instead of simply displaying the name of the center, would it be possible to add a link that redirects users to the address of the center or a page where they can find more details about the stamping center?

Reply:
Hi Rutuja, I have added the link. But the link is leading to National Park page as the available url is leading there. I will see if theres an alternative option or stamping center page but could add this url for now.